question.php

```diff
@@ -261,35 +311,45 @@ public function get_validation_error(array $response) {
      * @throws moodle_exception
      */
     public function grade_response(array $response): array {
-        $attemptscored = $this->api->package($this->packagehash, $this->packagefile)->score_attempt(
-            $this->questionstate,
-            $this->attemptstate,
-            $this->scoringstate,
-            $response
-        );
-        $this->update_ui($attemptscored->ui);
+        global $PAGE;
+
+        try {
+            $attemptscored = $this->api->package($this->packagehash, $this->packagefile)->score_attempt(
+                $this->questionstate,
+                $this->attemptstate,
+                $this->scoringstate,
+                $response
+            );
+            $this->update_ui($attemptscored->ui);
+        } catch (Throwable $t) {
+            // Trigger error event.
+            $params = [
+                'context' => $PAGE->context,
+                // TODO: It would be nice to set a 'relateduserid'.
+                'other' => [
+                    'questionid' => $this->id,
+                    'errormessage' => $t->getMessage(),
+                ],
+            ];
+            $event = \qtype_questionpy\event\grading_response_failed::create($params);
+            $event->trigger();
+            debugging($event->get_description());
+
+            // As the server was not able to score the response, we mark this question with manual scoring.
+            return [0, question_state::$needsgrading];
```

Comment:
An die User-ID könntest du mit `$this->get_behaviour()->get_pending_step()->get_user_id()` kommen, wobei ich nicht weiß, ob das vielleicht ein bisschen unsauber ist.